src/Altinn.App.Api/Models/InstanceFileScanResult.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+using Altinn.App.Core.Models;
+using Altinn.Platform.Storage.Interface.Enums;
+
+namespace Altinn.App.Api.Models
+{
+    /// <summary>
+    /// Light weight model representing an instance and it's file scan result status.
+    /// </summary>
+    public class InstanceFileScanResult
+    {
+        private readonly InstanceIdentifier _instanceIdentifier;
+        private readonly List<DataElementFileScanResult> _dataElements;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InstanceFileScanResult"/> class.
+        /// </summary>
+        public InstanceFileScanResult(InstanceIdentifier instanceIdentifier)
+        {
+            _instanceIdentifier = instanceIdentifier;
+            _dataElements = new List<DataElementFileScanResult>();
+        }
+
+        /// <summary>
+        /// Instance id
+        /// </summary>
+        [JsonPropertyName("id")]
+        public string Id
+        {
+            get { return _instanceIdentifier.ToString(); }
+        }
+
+        /// <summary>
+        /// Contains the aggregated file scan result for an instance.
+        /// Infected = If any data elements has a status of Infected
+        /// Clean = If all data elements has status Clean
+        /// Pending = If all or some are Pending and the rest are Clean
+        /// </summary>
+        [JsonPropertyName("fileScanResult")]
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        public FileScanResult FileScanResult { get; private set; }
+
+        /// <summary>
+        /// File scan result for individual data elements.
+        /// </summary>
+        [JsonPropertyName("data")]
+        public IReadOnlyList<DataElementFileScanResult> DataElements => _dataElements.AsReadOnly();
+
+        /// <summary>
+        /// Adds a individual data element file scan result and updates the aggregated file scan result status
+        /// </summary>        
+        public void AddFileScanResult(DataElementFileScanResult dataElementFileScanResult)
+        {
+            if (dataElementFileScanResult.FileScanResult != FileScanResult.NotApplicable)
+            {
+                _dataElements.Add(dataElementFileScanResult);
+
+                RecalculateAggregatedStatus();
+            }            
+        }
+
+        private void RecalculateAggregatedStatus()
+        {
+            if (_dataElements.TrueForAll(dataElement => dataElement.FileScanResult == FileScanResult.Clean))
+            {
+                FileScanResult = FileScanResult.Clean;
+            }
```

Comment:
Small performance improvement possible?
Assume FileScanResult.Clean and set status to Infected/Pending if Any of them found?
This would require us to add logic for each new result type to not have it report Clean even if new status present.

Reply:
Done:)